uvicorn/protocols/websockets/websockets_sansio_impl.py

```diff
@@ -0,0 +1,389 @@
+            elif self.handshake_initiated:
+                self.send_500_response()
+                self.queue.put_nowait({"type": "websocket.disconnect", "code": 1006})
+            self.transport.close()
+
+    def handle_events(self) -> None:
+        for event in self.conn.events_received():
+            if isinstance(event, Request):
+                self.handle_connect(event)
+            if isinstance(event, Frame):
+                if event.opcode == websockets.frames.Opcode.CONT:
+                    self.handle_cont(event)
+                elif event.opcode == websockets.frames.Opcode.TEXT:
+                    self.handle_text(event)
+                elif event.opcode == websockets.frames.Opcode.BINARY:
+                    self.handle_bytes(event)
+                elif event.opcode == websockets.frames.Opcode.PING:
+                    self.handle_ping(event)
+                elif event.opcode == websockets.frames.Opcode.PONG:
+                    self.handle_pong(event)
+                elif event.opcode == websockets.frames.Opcode.CLOSE:
+                    self.handle_close(event)
+
+    # Event handlers
+
+    def handle_connect(self, event: Request) -> None:
+        self.request = event
+        self.response = self.conn.accept(event)
+        self.handshake_initiated = True
+        # if status_code is not 101 return response
+        if self.response.status_code != 101:
+            self.handshake_complete = True
+            self.close_sent = True
+            self.conn.send_response(self.response)
+            output = self.conn.data_to_send()
+            self.transport.writelines(output)
+            self.transport.close()
+            return
+
+        headers = [
+            (key.encode("ascii"), value.encode("ascii", errors="surrogateescape"))
+            for key, value in event.headers.raw_items()
+        ]
+        raw_path, _, query_string = event.path.partition("?")
+        self.scope: "WebSocketScope" = {
+            "type": "websocket",
+            "asgi": {"version": self.config.asgi_version, "spec_version": "2.3"},
+            "http_version": "1.1",
+            "scheme": self.scheme,
+            "server": self.server,
+            "client": self.client,
+            "root_path": self.root_path,
+            "path": unquote(raw_path),
+            "raw_path": raw_path.encode("ascii"),
+            "query_string": query_string.encode("ascii"),
+            "headers": headers,
+            "subprotocols": event.headers.get_all("Sec-WebSocket-Protocol"),
+            "state": self.app_state.copy(),
+        }
+        self.queue.put_nowait({"type": "websocket.connect"})
+        task = self.loop.create_task(self.run_asgi())
+        task.add_done_callback(self.on_task_complete)
+        self.tasks.add(task)
+
+    def handle_cont(self, event: Frame) -> None:
+        self.bytes += event.data
+        if event.fin:
+            self.send_receive_event_to_app()
```

Comment:
These lines(195-197) are not covered due to the same reason as I mentioned for line no. 149